src/azure-api-utils.ts

```diff
@@ -1,9 +1,17 @@
-export function parseResource(armId: string): { resourceGroupName: string | undefined; name: string | undefined } {
-    // /subscriptions/{subid}/resourcegroups/{group}/providers/.../{name}
+export function parseResource(armId: string): {
+    parentResourceId: string | undefined;
+    subscriptionId: string | undefined;
+    resourceGroupName: string | undefined;
+    name: string | undefined;
+} {
+    // General armId Format: /subscriptions/{subid}/resourcegroups/{group}/providers/.../{name}
+    // armId format of a member cluster, given parentResource is a Fleet {parentResourceId}/members/{name}
     const bits = armId.split("/").filter((bit) => bit.length > 0);
     const resourceGroupName = bits[3];
+    const subscriptionId = bits[1];
     const name = bits[bits.length - 1];
-    return { resourceGroupName, name };
+    const parentResourceId = `/${bits.slice(0, bits.length - 2).join("/")}`; // take all except for the last two bits
```

Comment:
I feel this parseResource can be written better, given the PR size and time, we can do this as follow up or incremental.,